### PR TITLE
Sh/datalib

### PR DESCRIPTION
**Pull request by sirahd**
- remove `datalib` dependency to use `d3-time` and `d3-time-format` instead
- update type definition to reflect newer vega-lite version
- update some previous examples to use vega-lite 2.0

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -37,15 +38,20 @@
     "d3-collection": "^1.0.2",
     "d3-selection": "^1.0.4",
     "datalib": "^1.7.3",
```
Comment (domoritz) [inline package.json]: remove this?

File: src/index.ts [hunk 2]
```diff
@@ -1,40 +1,42 @@
...
 import {TEMPORAL} from 'vega-lite/build/src/type';
 import * as vl from 'vega-lite/build/src/vl';
+import {FieldOption, Option} from './options';
+import {SupplementedFieldOption} from './supplementedFieldOption';
 
 // by default, delay showing tooltip for 100 ms
-var DELAY = 100;
-var tooltipPromise: number = undefined;
-var tooltipActive = false;
+let DELAY = 100;
```
Comment (domoritz) [inline src/index.ts]: why not const?

File: src/index.ts [hunk 3]
```diff
@@ -1,40 +1,42 @@
...
+let tooltipPromise: number = undefined;
+let tooltipActive = false;
 
 export type VgView = any;
 type SceneGraph = {
   datum: {
-    _facetID: any,
-    _id: any
+    _facetID: number,
```
Comment (domoritz) [inline src/index.ts]: 👍

File: src/index.ts [hunk 4]
```diff
@@ -140,13 +142,13 @@ function cancelPromise() {
...
 
-/* d3mapping from fieldDef.type to formatType */
-var formatTypeMap: {[type: string]: 'number' | 'time'} = {
-  "quantitative": "number",
-  "temporal": "time",
-  "ordinal": undefined,
-  "nominal": undefined
-}
+/* timemapping from fieldDef.type to formatType */
+let formatTypeMap: {[type: string]: 'number' | 'time'} = {
```
Comment (domoritz) [inline src/index.ts]: const?

File: src/index.ts [hunk 5]
```diff
@@ -162,32 +164,30 @@ var formatTypeMap: {[type: string]: 'number' | 'time'} = {
 */
 function supplementOptions(options: Option, vlSpec: TopLevelExtendedSpec) {
   // fields to be supplemented by vlSpec
-  var supplementedFields: FieldOption[] = [];
+  let supplementedFields: FieldOption[] = [];
```
Comment (domoritz) [inline src/index.ts]: const?

Try to replace all let with const and then only change the ones that need to be changed.

File: src/index.ts [hunk 6]
```diff
@@ -207,36 +207,33 @@ function supplementOptions(options: Option, vlSpec: TopLevelExtendedSpec) {
...
+  }
 
   // if aggregate, match field name and aggregate operation
   if (fieldDef.aggregate) {
     // try find the perfect match: field name equals, aggregate operation equals
-    for (var i = 0; i < fieldOptions.length; i++) {
-      var fieldOption = fieldOptions[i];
-      if (fieldOption.field === fieldDef.field && fieldOption.aggregate === fieldDef.aggregate) {
-        return fieldOption;
+    for (let item of fieldOptions) {
```
Comment (domoritz) [inline src/index.ts]: 👍

File: src/index.ts [hunk 7]
```diff
@@ -688,57 +692,82 @@ function dropFieldsForLineArea(marktype: string, itemData: Map<string>) {
...
-  switch (dl.type(value)) {
-    case "date":
-      return dl.format.auto.time()(value);
-    case "number":
-      return dl.format.auto.number()(value);
-    case "boolean":
-    case "string":
-    default:
-      return value;
+function autoFormat(value: string | number | Date) {
```
Comment (domoritz) [inline src/index.ts]: return value is a string, right? Maybe just add it to make it explicit.

File: package.json (hunk 1 above)
Comment (sirahd) [inline package.json]: We still need `d3-collection` and `d3-selection` because of d3 map, but I plan to remove that to use javascript map instead

Comment (domoritz) [inline package.json]: Oh, but you can remove `datalib`

File: src/index.ts (hunk 7 above)
Comment (sirahd) [inline src/index.ts]: Style question: should I add explicit return type to every function?

**Pull request merged**